src/Stratis.Bitcoin.Features.PoA/PoANetwork.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using NBitcoin.DataEncoders;
+
+namespace Stratis.Bitcoin.Features.PoA
+{
+    public class PoANetwork : Network
+    {
+        /// <summary> The name of the root folder containing the different PoA blockchains.</summary>
+        private const string NetworkRootFolderName = "poa";
+
+        /// <summary> The default name used for the Stratis configuration file. </summary>
+        private const string NetworkDefaultConfigFilename = "poa.conf";
+
+        public PoANetwork()
+        {
+            // The message start string is designed to be unlikely to occur in normal data.
+            // The characters are rarely used upper ASCII, not valid as UTF-8, and produce
+            // a large 4-byte int at any alignment.
+            var messageStart = new byte[4];
+            messageStart[0] = 0x76;
+            messageStart[1] = 0x36;
+            messageStart[2] = 0x23;
+            messageStart[3] = 0x06;
+            uint magic = BitConverter.ToUInt32(messageStart, 0);
+
+            this.Name = "PoAMain";
+            this.Magic = magic;
+            this.DefaultPort = 16438;
+            this.RPCPort = 16474;
+            this.MaxTipAge = 2 * 60 * 60;
+            this.MinTxFee = 10000;
+            this.FallbackFee = 10000;
+            this.MinRelayTxFee = 10000;
+            this.RootFolderName = NetworkRootFolderName;
+            this.DefaultConfigFilename = NetworkDefaultConfigFilename;
+            this.MaxTimeOffsetSeconds = 25 * 60;
+            this.CoinTicker = "POA";
+
+            var consensusFactory = new PosConsensusFactory(); // TODO POA create new consensus factory that creates blocks with PoAHeaders
+
+            // Create the genesis block.
+            this.GenesisTime = 1513622125;
+            this.GenesisNonce = 1560058197;
+            this.GenesisBits = 402691653;
+            this.GenesisVersion = 1;
+            this.GenesisReward = Money.Zero;
+
+            Block genesisBlock = CreatePoAGenesisBlock(consensusFactory, this.GenesisTime, this.GenesisNonce, this.GenesisBits, this.GenesisVersion, this.GenesisReward);
+
+            this.Genesis = genesisBlock;
+
+            var consensusOptions = new PosConsensusOptions(
```

Comment:
You are probably gonna need a `PoaConsensusOptions` with different params (next PR)